cmd/dep/gom_importer.go

```diff
@@ -0,0 +1,268 @@
+				it := strings.TrimSpace(as[i][0])
+				if strings.HasPrefix(it, ",") {
+					it = strings.TrimSpace(it[1:])
+				}
+				if strings.HasPrefix(it, ":") {
+					it = strings.TrimSpace(it[1:])
+				}
+				a = append(a, it)
+			}
+			options[kvs[0][1:]] = a
+		} else {
+			options[kvs[0][1:]] = unquote(kvs[1])
+		}
+	}
+}
+
+// load the gomfile.
+func (g *gomImporter) load(projectDir string) error {
+	g.logger.Println("Detected Gomfile...")
+	filename := filepath.Join(projectDir, gomfileName)
+	f, err := os.Open(filename + ".lock")
+	if err != nil {
+		f, err = os.Open(filename)
+		if err != nil {
+			return err
+		}
+	}
+	defer f.Close()
+	br := bufio.NewReader(f)
+
+	g.goms = make([]gomPackage, 0)
+
+	n := 0
+	skip := 0
+	valid := true
+	var envs []string
+	for {
+		n++
+		lb, _, err := br.ReadLine()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		line := strings.TrimSpace(string(lb))
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		name := ""
+		options := make(map[string]interface{})
+		var items []string
+		if gomReGroup.MatchString(line) {
+			envs = strings.Split(gomReGroup.FindStringSubmatch(line)[1], ",")
+			for i := range envs {
+				envs[i] = strings.TrimSpace(envs[i])[1:]
+			}
+			valid = true
+			continue
+		} else if gomReEnd.MatchString(line) {
+			if !valid {
+				skip--
+				if skip < 0 {
+					return fmt.Errorf("Syntax Error at line %d", n)
+				}
+			}
+			valid = false
+			envs = nil
+			continue
+		} else if skip > 0 {
+			continue
+		} else if gomReGom.MatchString(line) {
+			items = gomReGom.FindStringSubmatch(line)[1:]
+			name = unquote(items[0])
+			g.parseOptions(items[1], options)
+		} else {
+			return fmt.Errorf("Syntax Error at line %d", n)
+		}
+		if envs != nil {
+			options["group"] = envs
+		}
+		g.goms = append(g.goms, gomPackage{name, options})
+	}
+}
+
+// convert the gomfile into dep configuration files.
+func (g *gomImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from Gomfile ...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.goms {
+		// Obtain ProjectRoot. Required for avoiding sub-package imports.
+		ip, err := g.sm.DeduceProjectRoot(pkg.name)
+		if err != nil {
+			return nil, nil, err
+		}
+		pkg.name = string(ip)
+
+		// Check if it already existing in locked projects
+		if projectExistsInLock(lock, pkg.name) {
+			continue
+		}
+
+		rev := ""
+
+		if pkg.hasOption("branch") {
+			rev, _ = pkg.options["branch"].(string)
+		}
+		if pkg.hasOption("tag") {
+			rev, _ = pkg.options["tag"].(string)
+		}
+		if pkg.hasOption("commit") {
+			rev, _ = pkg.options["commit"].(string)
+		}
+
+		var pc gps.ProjectConstraint
+		pi := gps.ProjectIdentifier{
+			ProjectRoot: gps.ProjectRoot(pkg.name),
+		}
+
+		pc.Ident = pi
+
+		if rev != "" {
+			pc.Constraint, err = g.sm.InferConstraint(rev, pc.Ident)
```

Comment:
After thinking about this for a bit, I think the right answer is to do that in `InferConstraint`, so that when an empty string is passed, it returns `gps.Any()`. I'll submit a PR for that in a bit.